### PR TITLE
Add FP64 XDL GEMM built-in function

### PR DESCRIPTION
**Pull request by ltqin**

### CODE REVIEW

**Comment by asroy** [comment]
@ltqin CI failed

**Comment by ltqin** [comment]
> @ltqin CI failed

@asroy @zjing14 CI pass

File: example/01_gemm/gemm_xdl_fp64.cpp [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+//##########|      |      |      |        |        |        |        |            |            |            |              |      |      |      |      |   |     |     |     |     |                |               |               |               |               |               |          |                |               |               |              |               |               |          |                |                |
+             <  F64,   F64,   F64,     F64,     Row,     Col,     Row, PassThrough, PassThrough, PassThrough,   GemmDefault,   64,    32,    32,     4,  1,   16,   16,    2,    2,     S<4, 16, 1>,     S<1, 0, 2>,     S<1, 0, 2>,              2,              1,              1,      true,     S<4, 16, 1>,     S<1, 0, 2>,     S<1, 0, 2>,             2,              1,              1,      true,               7,               1>;
```
Comment (zjing14) [inline example/01_gemm/gemm_xdl_fp64.cpp:54]: @ltqin Did you try larger tile size, i.e., 128x128 or 128x256?

Comment (ltqin) [reply example/01_gemm/gemm_xdl_fp64.cpp:55]: done

**Comment by asroy** [comment]
@ltqin CI failed

**Comment by ltqin** [comment]
@asroy the test can pass on rocm5.0 and 9110, but not failed on rocm5.1

**Review by asroy — APPROVED** [review]

**Pull request merged**